### PR TITLE
Feature/add user data object

### PR DESCRIPTION
**Pull request by 0xLeif**

### CODE REVIEW

File: Sources/App/Models/User.swift [hunk 1]
```diff
@@ -9,6 +9,10 @@ import FluentSQLite
 import Vapor
 import Authentication
 
+struct UserData: Content {
+    let id: Int?
```
Comment (will-nemo) [inline Sources/App/Models/User.swift:13]: Will we be adding more fields in the future for the UserData Object?

Comment (0xLeif) [reply Sources/App/Models/User.swift:13]: We might. I'm going to clean this up quick.

Comment (0xLeif) [reply Sources/App/Models/User.swift:13]:
```
struct UserData: Content {
    let username: String
}

final class User: SQLiteModel {
    var id: Int?
    var userData: UserData
    var password: String
    
    init(id: Int? = nil, username: String, password: String) {
        self.id = id
        self.userData = UserData(username: username)
        self.password = password
    }
}

- extension User {
-     var userData: UserData {
-         return UserData(id: id, username: username)
-     }
- }
```

Comment (will-nemo) [reply Sources/App/Models/User.swift:13]: We'll be storing their password? Will we be hashing it ourselves?

Comment (0xLeif) [reply Sources/App/Models/User.swift:13]: The password is already stored and hashed. The UserData object is used so we can send data about the user without sending their hashed password in the object.

**Review by will-nemo — APPROVED** [review]

**Review by ambid17 — APPROVED** [review]

**Pull request merged**